### ISSUE TITLE
Integration with BitBucket stopped to work after changing permissions to the repo.

### ISSUE THREAD

**Issue by gdemecki** [Shippable/support#3611]
Recently I've changed my public repository on the BitBucket to be private. Since then I'm encountering numerous problems.

I've got mail that my [build failed](https://app.shippable.com/bitbucket/gdemecki/netadmin/runs/77/1/console) but the link shows *404 Page not found* (even if I'm logged in). Why?

Anyway, in Account Settings -> Integrations, I've tried to modify/ remove/ re-create my integration with BitBucket but I'm encountering another problems.

On **Edit** I'm getting error:
```
(Id: 4004) No subscriptions found for subscriptionId(s):
verbose: /subscriptionIntegrations|GET|callerId:5805e712a708080e00ad01d5|validateSubscriptions|_computeShim
```

On **Delete** I'm getting error:
```
(Id: 4001) accountIntegrations.destroy failed for id:
verbose: accountIntegrations|deleteById|callerId:5805e712a708080e00ad01d5|accountIntegrationId:5926844c2811330700bf42e7|_deleteById
"5926844c2811330700bf42e7 returned error: update or delete on table \"accountIntegrations\" violates foreign key constraint \"subscriptionIntegrations_accountIntegrationId_fkey\" on table \"subscriptionIntegrations\""
```

IMHO UI for integrations is not very user friendly. I don't even remember how I was able to integrate Shippable with Bitbucket six months ago. I'm stuck. What I'm supposed to do?

Another problem is that I cannot _Enable a Project_ when I'm on the main Dashboard page. Again why?

EDIT:
Why even [Shippable documentation ](http://docs.shippable.com/tutorials/ci/scm-bitbucket-server-ci/) stopped to work?

Regards

**Comment by nranjanShippable** [comment]
Hi @gdemecki,
Did you sync your account after making the changes to your BitBucket repository. If yes, then are you able to see any of the subscription?.

If you haven't synced your account then you can re-sync in Account Settings -> Accounts and then sync.
After doing so, are you able to see the subscription?

If your are able to see the subscriptions, then try the same with your bitbucket integration. If the error persists, do inform us.

**Comment by gdemecki** [comment]
Hi @nranjanShippable 

Yes, I did Sync many times, it didn't help back then, but surprisingly UI now works.

But now I have another problem: I cannot generate  BitBucket API token, I'm always getting:
![screen](https://cloud.githubusercontent.com/assets/617283/26447620/9b15a8c4-4149-11e7-9077-4421e78382d9.png)

Do you happen to know what should I pass as a `redirect_uri`? Has something changed recently when it comes to OAuth and BitBucket?

**Comment by nranjanShippable** [comment]
@gdemecki 
We are looking into this, will get back to you with our findings.

**Comment by ambarish2012** [comment]
@gdemecki it seems that you have deleted your subscription and account integration.

This is what i would recommend -

1) Revoke the OAuth permission for Shippable from Bitbucket.

Go to your profile in Bitbucket -> Click on OAuth -> Click on Revoke for Shippable.

2) Logout of Shippable.

3) Login into Shippable with your bitbucket account.

4) This will cleanly recreate your bitbucket account integration with the right token. Go to account integration and you should see the bitbucket SCM integration created.

If you run into errors with step 4, let me know.

**Comment by ambarish2012** [comment]
@gdemecki was your problem resolved?

**Comment by gdemecki** [comment]
@ambarish2012 Hi, sorry I was busy last week.
So it seems that 'Logout' step caused that I was asked to grant permissions again, and It solved all errors on the UI. So far so good.

But I've launched build one more time, I've got old, well known error message during `git_sync`step:
```
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I've checked one more time my BitBucket's repository settings. I see "Access Key" for Shippable already added, also page with `OAuth` settings shows `Shippable` as an authorized application.
So why Shippable cannot sync?

**Comment by ambarish2012** [comment]
@gdemecki please can you send me the link to your console? Also - would you be available for a quick google hangout / zoom session so that we can look at your bitbucket configuration?

**Comment by ambarish2012** [comment]
@gdemecki are you running bitbucket server?

**Comment by ambarish2012** [comment]
@gdemecki - your private repository most likely does not have the Shippable deploy key, which is needed to sync the repository.

To create the deploy SSH key, please follow these steps -

1) Navigate to https://app.shippable.com/bitbucket/gdemecki/netadmin/settings/options
2) Click on the `Reset` button.
3) Run the build.

**Comment by gdemecki** [comment]
@nranjanShippable, @ambarish2012  You've made my day! That's solved my last problem.

Thanks for your excellent support, you've been able to solve this issue even without our hangout session!

But when it comes to the Shippable service itself, I cannot say such good words. It's not intuitive that I had to do the "logout" just to make Shippable UI work back again.
The same goes for the "Reset" button. Shouldn't  "deploy access key" be re-created automatically?

Another thing which was strange - Shippable had enough permissions to mark my commits in the BitBucket GUI as "FAILED" (I'm talking about this red icon next to the commit date), but at the same time, it couldn't access the needed repository? That sounds like a nonsense. Moreover, literally a minute ago, I've granted *all* permissions to my BitBucket account (see #3410).

To sum it up, I wouldn't be able to solve it, without your explicit support...
